src/AppBundle/Entity/LegislativeDistrictZone.php

```diff
@@ -139,11 +144,40 @@ public function getAreaCode(): ?string
 
     public function setAreaCode(string $code): void
     {
-        $this->areaCode = $code;
+        $this->areaCode = sprintf('%04s', $code);
+
+        if (null === $this->rank) {
+            $this->rank = self::guessRankFromAreaCode($code);
+        }
+    }
+
+    private static function guessRankFromAreaCode(string $areaCode): int
```

Comment:
No need to be static I guess.
Maybe you should explain why you add rank+1 after corse area in a phpdoc.